### PR TITLE
switch to animation for opacity

### PR DESCRIPTION
**Pull request by jonahwilliams**
To mirror how Image works from flutter/widgets.

### CODE REVIEW

File: packages/vector_graphics/test/render_vector_graphics_test.dart [hunk 1]
```diff
@@ -195,3 +222,41 @@ class FakePaintingContext extends Fake implements PaintingContext {
   @override
   final FakeCanvas canvas = FakeCanvas();
 }
+
+class FixedOpacityAnimation extends Animation<double> {
```
Comment (dnfield) [inline packages/vector_graphics/test/render_vector_graphics_test.dart:226]: Use https://api.flutter.dev/flutter/animation/AlwaysStoppedAnimation-class.html

File: packages/vector_graphics/lib/vector_graphics.dart [hunk 2]
```diff
@@ -114,7 +113,7 @@ class VectorGraphic extends StatefulWidget {
   /// This value does not apply to the widgets created by a [placeholderBuilder].
   /// Using this value is more efficient than wrapping this widget in an
   /// [Opacity] or [FadeTransition].
-  final double opacity;
+  final Animation<double>? opacity;
```
Comment (dnfield) [inline packages/vector_graphics/lib/vector_graphics.dart:116]: See the docs on https://api.flutter.dev/flutter/widgets/Image/opacity.html

We should mention about using `AlwaysStoppedAnimation` to provide a fixed value.

File: packages/vector_graphics/test/render_vector_graphics_test.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/vector_graphics/test/render_vector_graphics_test.dart:226]: TIL, thanks!

Comment (jonahwilliams) [reply packages/vector_graphics/test/render_vector_graphics_test.dart:226]: actually, I want this class for the test so I can verify the listeners

File: packages/vector_graphics/lib/vector_graphics.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/vector_graphics/lib/vector_graphics.dart:136]: Updated the docs to mention AlwaysStoppedAnimation

File: packages/vector_graphics/test/render_vector_graphics_test.dart (hunk 1 above)
Comment (dnfield) [reply packages/vector_graphics/test/render_vector_graphics_test.dart:226]: Fair

**Review by dnfield — APPROVED** [review]

**Pull request merged**